pages/api/coupon-fraud/claim.js

```diff
@@ -0,0 +1,78 @@
+import { getForbiddenReponse, getOkReponse } from '../../../server/server';
+import { Op } from 'sequelize';
+import { couponEndpoint } from '../../../server/coupon-fraud/coupon-endpoint';
+import { CouponClaim, CouponCode } from '../../../server/coupon-fraud/database';
+
+async function checkVisitorClaimedRecently(visitorId) {
+  const oneHourBefore = new Date();
+  oneHourBefore.setHours(oneHourBefore.getHours() - 1);
+
+  return await CouponClaim.findOne({
+    where: {
+      visitorId,
+      timestamp: {
+        [Op.between]: [oneHourBefore, new Date()],
+      },
+    },
+  });
+}
+
+async function getVisitorClaim(visitorId, couponCode) {
+  return await CouponClaim.findOne({
+    where: { visitorId, couponCode },
+  });
+}
+
+/**
+ * Checks if given visitor has already used a coupon code.
```

Comment:
This comment (or function) is not correct. There's no visitorId in the function.